src/components/Settings.jsx

```diff
@@ -97,10 +107,56 @@ export default function Settings() {
   const isSats = settings.unit === SATS
   const isLightTheme = settings.theme === window.JM.THEMES[0]
 
+  const lockWallet = useCallback(
+    async ({ force, destination }) => {
+      if (!force && (serviceInfo.coinjoinInProgress || serviceInfo.makerRunning)) {
+        setShowConfirmLockModal({ destination })
+        return
+      }
+
+      setLockingWallet(true)
+
+      try {
+        const { name: walletName, token } = currentWallet
+        const res = await Api.getWalletLock({ walletName, token })
+
+        if (res.ok || res.status === 401) {
+          stopWallet()
+        } else if (!res.ok) {
+          Api.Helper.throwError(res)
+        }
+
+        navigate(destination)
+      } catch (e) {
+        setAlert({ variant: 'danger', dismissible: false, message: e.message })
+      } finally {
+        setLockingWallet(false)
+      }
```

Comment:
```suggestion
        setLockingWallet(false)
        navigate(destination)
      } catch (e) {
        setLockingWallet(false)
        setAlert({ variant: 'danger', dismissible: false, message: e.message })
      }
```

Prevents `Can't perform a React state update on an unmounted component`.